providers/redis/shedlock-provider-redis-quarkus2/src/main/java/net/javacrumbs/shedlock/provider/redis/quarkus2/QuarkusRedisLockProvider.java

```diff
@@ -0,0 +1,145 @@
+package net.javacrumbs.shedlock.provider.redis.quarkus2;
+
+import static net.javacrumbs.shedlock.support.Utils.getHostname;
+import static net.javacrumbs.shedlock.support.Utils.toIsoString;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+
+import io.quarkus.logging.Log;
+import io.quarkus.redis.datasource.RedisDataSource;
+import io.quarkus.redis.datasource.keys.KeyCommands;
+import io.quarkus.redis.datasource.value.SetArgs;
+import io.quarkus.redis.datasource.value.ValueCommands;
+import io.quarkus.runtime.configuration.ConfigUtils;
+import net.javacrumbs.shedlock.core.AbstractSimpleLock;
+import net.javacrumbs.shedlock.core.ClockProvider;
+import net.javacrumbs.shedlock.core.ExtensibleLockProvider;
+import net.javacrumbs.shedlock.core.LockConfiguration;
+import net.javacrumbs.shedlock.core.SimpleLock;
+import net.javacrumbs.shedlock.support.LockException;
+import net.javacrumbs.shedlock.support.annotation.NonNull;
+
+/**
+ * Uses Redis's `SET resource-name anystring NX EX max-lock-ms-time` as locking mechanism.
+ * <p>
+ * See <a href="https://redis.io/commands/set">Set command</a>
+ */
+public class QuarkusRedisLockProvider implements ExtensibleLockProvider {
+
+    private static final String KEY_PREFIX = "lock";
+
+    private final RedisDataSource redisDataSource;
+    private final ValueCommands<String, String> valueCommands;
+    private final KeyCommands<String> keyCommands;
+    private final String environment;
+
+    public QuarkusRedisLockProvider(RedisDataSource dataSource, String appNameOrPrefix) {
+        this.redisDataSource = dataSource;
+        this.environment = appNameOrPrefix + ":"+ String.join(":", ConfigUtils.getProfiles());
```

Comment:
Can you please explain why the enviroment is used for the key. I do not think we have anything like this in any other lock provider.

Reply:
Ditto the appNameOrPrefix. If you have a use-case for it, let's just keep the `appNameOrPrefix`.

Reply:
Part of this code was copied from the spring implementation, it used the profile/environment name to compose the key. I ended up leaving it as it was.